test/framework/options.py

```diff
@@ -776,6 +776,49 @@ def test_search(self):
                 args = [opt, pattern, '--robot', test_easyconfigs_dir]
                 self.assertErrorRegex(EasyBuildError, "Invalid search query", self.eb_main, args, raise_error=True)
 
+    def test_ignore_index(self):
+        """
+        Test use of --ignore-index.
+        """
+
+        test_ecs_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'easyconfigs')
+        toy_ec = os.path.join(test_ecs_dir, 'test_ecs', 't', 'toy', 'toy-0.0.eb')
+        copy_file(toy_ec, self.test_prefix)
+
+        # install index that list more files than are actually available,
+        # so we can check whether it's used
+        index_txt = '\n'.join([
+            'toy-0.0.eb',
+            'toy-1.2.3.eb',
+            'toy-4.5.6.eb',
+        ])
+        write_file(os.path.join(self.test_prefix, '.eb-path-index'), index_txt)
```

Comment:
Because then we wouldn't easily catch a change to the `PATH_INDEX_FILENAME` constant.

I'm deliberately hardcoding the `.eb-path-index` here, since if the constant would be changed for whatever weird reason, that should make the tests fail (since it changes the EasyBuild behavior, existing indexes would suddenly be ignored).